src/common/uriHelpers.ts

```diff
@@ -70,6 +79,7 @@ export function parseUriCommon(
   }
 
   const amountStr = parsedUri.query.amount
+  let currencyCode = currencyCodeArg
```

Comment:
Maybe move this up to the top, and make it `let {currencyCode} = opts`. Then we can drop the weird `currencyCodeArg` thing.